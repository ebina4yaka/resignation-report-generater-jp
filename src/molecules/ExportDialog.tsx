import React from 'react'
import { createStyles, makeStyles } from '@material-ui/core/styles'
import Dialog from '@material-ui/core/Dialog'
import { TransitionProps } from '@material-ui/core/transitions'
import Slide from '@material-ui/core/Slide'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogTitle from '@material-ui/core/DialogTitle'
import ExportDialogButtons from './ExportDialogButtons'
import DisplayParameters from './DisplayParameters'

type Props = {
  isOpen: boolean
  handleClose: () => void
}

const useStyles = makeStyles(
  createStyles({
    buttonArea: {
      display: 'flex',
      flexWrap: 'wrap',
      flexDirection: 'row',
      justifyContent: 'center',
    },
  })
)

const Transition = React.forwardRef(function Transition(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any,react/require-default-props
  props: TransitionProps & { children?: React.ReactElement<any, any> },
  ref: React.Ref<unknown>
) {
  // eslint-disable-next-line react/jsx-props-no-spreading
  return <Slide direction="up" ref={ref} {...props} />
})

export default function ExportDialog(props: Props): React.ReactElement {
  const { isOpen, handleClose } = props
  const classes = useStyles()
  return (
    <Dialog
      open={isOpen}
      TransitionComponent={Transition}
      keepMounted
      onClose={handleClose}
      aria-labelledby="alert-dialog-slide-title"
      aria-describedby="alert-dialog-slide-description"
    >
      <DialogTitle id="alert-dialog-slide-title">入力内容の確認</DialogTitle>
      <DialogContent>
        <DisplayParameters />
      </DialogContent>
      <DialogActions className={classes.buttonArea}>
        <ExportDialogButtons handleClose={handleClose} />
      </DialogActions>
    </Dialog>
  )
}
