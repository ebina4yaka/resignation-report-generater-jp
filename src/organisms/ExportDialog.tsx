import { forwardRef, Ref, ReactElement, useContext } from 'react'
import { createStyles, makeStyles } from '@material-ui/core/styles'
import DialogTitle from '@material-ui/core/DialogTitle'
import DialogContent from '@material-ui/core/DialogContent'
import DialogActions from '@material-ui/core/DialogActions'
import Dialog from '@material-ui/core/Dialog'
import { TransitionProps } from '@material-ui/core/transitions'
import Slide from '@material-ui/core/Slide'
import { parametersContext } from '../context/useParameters'
import ConfirmButton from '../atoms/ConfirmButton'
import useIsButtonDisable from '../hooks/useIsButtonDisable'
import useDialogOpen from '../hooks/useDialogOpen'
import PreviewDocxText from '../molecules/PreviewDocxText'
import ExportDialogButtons from '../molecules/ExportDialogButtons'
import IsPrintPaidLeaveRequestForm from '../atoms/IsPrintPaidLeaveRequestForm'

const useStyles = makeStyles((theme) =>
  createStyles({
    root: {
      display: 'flex',
      flexWrap: 'wrap',
      flexDirection: 'row',
      justifyContent: 'center',
    },
    title: {
      textAlign: 'center',
    },
    confirm: {
      marginBottom: theme.spacing(3),
    },
    buttonArea: {
      display: 'flex',
      flexWrap: 'wrap',
      flexDirection: 'row',
      justifyContent: 'center',
    },
  })
)

const Transition = forwardRef(function Transition(
  // eslint-disable-next-line react/require-default-props
  props: TransitionProps & { children?: ReactElement },
  ref: Ref<unknown>
) {
  // eslint-disable-next-line react/jsx-props-no-spreading
  return <Slide direction="up" ref={ref} {...props} />
})

export default function ExportDialog(): ReactElement {
  const context = useContext(parametersContext)
  const { daysOfPaidLeaveRemaining } = context
  const classes = useStyles()
  const { isOpen, handleOpen, handleClose } = useDialogOpen()
  return (
    <div className={classes.root}>
      <div className={classes.confirm}>
        <ConfirmButton
          text="確認"
          isDisable={useIsButtonDisable(context)}
          onClickFunction={handleOpen}
        />
      </div>
      <Dialog
        open={isOpen}
        TransitionComponent={Transition}
        keepMounted
        onClose={handleClose}
        fullWidth
        maxWidth="md"
      >
        <DialogTitle className={classes.title}>プレビュー</DialogTitle>
        <DialogContent>
          <PreviewDocxText />
          <IsPrintPaidLeaveRequestForm
            daysOfPaidLeaveRemaining={daysOfPaidLeaveRemaining}
          />
        </DialogContent>
        <DialogActions className={classes.buttonArea}>
          <ExportDialogButtons handleClose={handleClose} />
        </DialogActions>
      </Dialog>
    </div>
  )
}
