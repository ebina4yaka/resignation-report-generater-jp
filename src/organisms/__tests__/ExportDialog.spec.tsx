import renderer from 'react-test-renderer'
import ExportDialog from '../ExportDialog'

jest.mock('../../atoms/ConfirmButton', () => 'MockedConfirmButton')
jest.mock('../../molecules/PreviewDocxText', () => 'MockedPreviewDocxText')
jest.mock(
  '../../atoms/IsPrintPaidLeaveRequestForm',
  () => 'MockedIsPrintPaidLeaveRequestForm'
)
jest.mock(
  '../../molecules/ExportDialogButtons',
  () => 'MockedExportDialogButtons'
)
jest.mock('@material-ui/core/Dialog', () => 'MockedDialog')

test('Snapshot test for ExportDialog', () => {
  const tree = renderer.create(<ExportDialog />).toJSON()
  expect(tree).toMatchSnapshot()
})
