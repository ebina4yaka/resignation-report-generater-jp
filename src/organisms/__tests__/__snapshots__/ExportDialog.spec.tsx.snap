// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for ExportDialog 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="makeStyles-confirm-3"
  >
    <MockedConfirmButton
      isDisable={false}
      onClickFunction={[Function]}
      text="確認"
    />
  </div>
  <MockedDialog
    TransitionComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    fullWidth={true}
    keepMounted={true}
    maxWidth="md"
    onClose={[Function]}
    open={false}
  >
    <div
      className="MuiDialogTitle-root makeStyles-title-2"
    >
      <h2
        className="MuiTypography-root MuiTypography-h6"
      >
        プレビュー
      </h2>
    </div>
    <div
      className="MuiDialogContent-root"
    />
    <div
      className="MuiDialogActions-root makeStyles-buttonArea-4 MuiDialogActions-spacing"
    >
      <MockedExportDialogButtons
        handleClose={[Function]}
      />
    </div>
  </MockedDialog>
</div>
`;
